iothub/device/src/ClientPropertyCollection.cs

```diff
@@ -55,8 +63,16 @@ public void AddComponentProperty(string componentName, string propertyName, obje
         /// <param name="componentName">The component with the properties to add.</param>
         /// <param name="properties">A collection of properties to add.</param>
         /// <exception cref="ArgumentException">A property name in <paramref name="properties"/> already exists in the collection.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="componentName"/> or a property name in <paramref name="properties"/> is <c>null</c>.</exception>
         public void AddComponentProperties(string componentName, IDictionary<string, object> properties)
-            => AddInternal(properties, componentName, true);
+        {
+            if (componentName == null)
+            {
+                throw new ArgumentNullException(nameof(componentName));
+            }
+
+            AddInternal(properties, componentName, false);
```

Comment:
Should the final parameter here be ```true```? If not, why is it being changed from its previous behavior here?

Reply:
The final parameter here determines if the operation is a "force" update. This is set to false for the "Add" APIs and is set to true for the "AddOrUpdate" APIs.
This was incorrectly set to true previously.